cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -332,15 +333,16 @@ public Cluster kafkaCluster() {
    */
   public void pauseMetricSampling() {
     _loadMonitorTaskRunner.pauseSampling();
+    _loadMonitorTaskRunner.setAwaitExecution(true);
```

Comment:
`pauseMetricSampling ()` and `resumeMetricSampling ()` are not explicitly called by the `Executor` while starting an execution. Users can also use the REST API to resume/pause sampling. Hence, the assumption regarding `setAwaitExecution(true)` here and `setAwaitExecution(false)` in `resumeMetricSampling()` is incorrect.

A potential mitigation is to change this function's signature to get a `boolean` to indicate whether an execution is starting or not.